src/Bridge/Symfony/Routing/RouteNameGenerator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Routing;
+
+use Doctrine\Common\Util\Inflector;
+
+/**
+ * Generates the Symfony route name associated with an operation name and a resource short name.
+ *
+ * @author Baptiste Meyer <baptiste.meyer@gmail.com>
+ */
+class RouteNameGenerator
+{
+    const ROUTE_NAME_PREFIX = 'api_';
+
+    private function __construct()
+    {
+    }
+
+    /**
+     * Generates a Symfony route name.
+     *
+     * @param string $operationName
+     * @param string $resourceShortName
+     * @param bool   $collection
+     *
+     * @return string
+     */
+    public static function generate(string $operationName, string $resourceShortName, bool $collection): string
```

Comment:
Oh okay @dunglas didn't saw your comment, and I'm :+1: on merging this in master! So yes, please drop the `bool $collection` for an `OperationType` :D.